assets/app/scripts/directives/metrics.js

```diff
@@ -77,13 +107,19 @@ angular.module('openshiftConsole')
 
         scope.usageByMetric = {};
 
+        scope.anyUsageByMetric = function(metric) {
+          _.some(_.map(metric.datasets, 'id'), function(metricID) { scope.usageByMetric[metricID] !== undefined; });
+        };
+
         var createDonutConfig = function(metric) {
           var chartID = '#' + metric.chartPrefix + scope.uniqueID + '-donut';
           return {
             bindto: chartID,
             onrendered: function() {
-              var used = scope.usageByMetric[metric.id].used;
-              ChartsService.updateDonutCenterText(chartID, used, metric.units);
+              angular.forEach(_.map(metric.datasets, 'id'), function(metricID) {
+                var used = scope.usageByMetric[metricID].used;
+                ChartsService.updateDonutCenterText(chartID, used, metric.units);
+              });
```

Comment:
Since there's only one chart, I'd remove the loop and just use the first dataset value. (I realize memory/cpu have only one dataset anyway.)